packages/react/src/themes/base/components/Text/textVariables.ts

```
@@ -1,8 +1,7 @@
-import { ColorValues } from '../../../types'
-import { mapColorsToScheme } from '../../../../lib'
+import { ColorSchemeMapping } from '../../../types'
 
 export interface TextVariables {
-  colors: ColorValues<string>
+  colorScheme: ColorSchemeMapping
```

this common type implies that supported `ColorNames` (in case if `color` props is supported by component) should be the same for any component of the app. In other words, if there is `pink` in color names, it should be supported by all components in corresponding schemes.

Won't it be a problem if, for instance, we would need to support extended range of colors for some particular component? Or this case is not considered intentional, as design spec rules explicitly won't allow this to happen? What would be our options to support this case, if necessary?

I see `extendColorScheme` method, so I might expect that we will introduce only minimal set of color options for base scheme, and will allow components extend this set of necessary, without affecting others - am I right on that?

---

also, related question: currently, for any component that supports `color` prop, it is possible to introduce the following markup :

```jsx
<Text color={any_color_from_color_names} />
<Segment ... > // same thing
```

Wondering of we would like to introduce this much flexibility to the client - as, essentially, only subset of these color variants of `Text` (or `Segment`) components is supposed to be used in the app.

Of course, client might omit defining colors, and then default one will be applied on render - but then markup will be misleading.

Could you, please, suggest our plans to handle this situations?

---

Very good question. As part of this PR, I just scoped this color value for the prop, to the colors we actually have, and I am defaulting this to the default styles if the color is not in this set. Ideally, we would have color values available defined per component, but this was still not specified by design. After we have these specs, we can update the component styles to handle only the colors they are suppose to. These changes were indeed just to unify the colors, and return default styles if something not expected is provided.

---

got it, thanks!